draft-irtf-cfrg-hpke.md

```diff
@@ -374,16 +382,18 @@ TLS presentation syntax:
 struct {
     // Mode and algorithms
     uint8 mode;
-    uint16 ciphersuite;
+    uint16 kem_id;
+    uint16 kdf_id;
+    uint16 aead_id;
 
     // Public inputs to this key exchange
     opaque enc[Nenc];
     opaque pkR[Npk];
     opaque pkI[Npk];
     opaque pskID<0..2^16-1>;
 
-    // Application-supplied info
-    opaque info<0..2^16-1>;
+    // Cryptographic hash of application-supplied info
+    opaque info_hash<0..512>;
```

Comment:
It seems like this could be made fixed-length, of length Nh:

```
opaque info_hash[Nh];
```

And likewise, drop the `len(info_hash)` above.

Reply:
Also, we now have only one variable-length field in the context, `pskID`.  Maybe we should hash that as well?  Maybe that's a follow-on.

Reply:
That's probably a good idea to get rid of variable-size fields.